src/TrinaryLogic.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace OndraM\CiDetector;
+
+/**
+ * @see https://en.wikipedia.org/wiki/Three-valued_logic
+ * @see https://github.com/phpstan/phpstan-src/blob/cedc99f51f7b8d815036e983166d7cb51ab46734/src/TrinaryLogic.php
+ *
+ * @internal
+ */
+final class TrinaryLogic
+{
+    private const YES = 1;
+    private const MAYBE = 0;
+    private const NO = -1;
+
+    /** @var self[] */
+    private static $registry = [];
+
+    /** @var int */
+    private $value;
+
+    private function __construct(int $value)
+    {
+        $this->value = $value;
+    }
+
+    public static function createMaybe(): self
+    {
+        return self::create(self::MAYBE);
+    }
+
+    public static function createFromBoolean(bool $value): self
+    {
+        return self::create($value ? self::YES : self::NO);
+    }
+
+    private static function create(int $value): self
+    {
+        self::$registry[$value] = self::$registry[$value] ?? new self($value);
+
+        return self::$registry[$value];
+    }
+
+    public function yes(): bool
+    {
+        return $this->value === self::YES;
+    }
+
+    public function maybe(): bool
+    {
+        return $this->value === self::MAYBE;
+    }
+
+    public function no(): bool
+    {
+        return $this->value === self::NO;
+    }
+
+    public function equals(self $other): bool
```

Comment:
This seems a bit YAGNI, I suggest removing this method.

Reply:
Agree